src/core/Mixer.cpp

```diff
@@ -446,30 +438,25 @@ const surroundSampleFrame * Mixer::renderNextBuffer()
 
 	m_profiler.finishPeriod( processingSampleRate(), m_framesPerPeriod );
 
-	return m_readBuf;
+	return m_outputBufferRead;
 }
 
 
 
 
 void Mixer::swapBuffers()
 {
-	// swap buffer
 	m_inputBufferWrite = ( m_inputBufferWrite + 1 ) % 2;
 	m_inputBufferRead =  ( m_inputBufferRead + 1 ) % 2;
 
-	// clear new write buffer
 	m_inputBufferFrames[ m_inputBufferWrite ] = 0;
 
-	// rotate buffers
-	m_writeBuffer = ( m_writeBuffer + 1 ) % m_poolDepth;
-	m_readBuffer = ( m_readBuffer + 1 ) % m_poolDepth;
-
-	m_writeBuf = m_bufferPool[m_writeBuffer];
-	m_readBuf = m_bufferPool[m_readBuffer];
+	surroundSampleFrame * outputBufferTemp = m_outputBufferRead;
+	m_outputBufferWrite = outputBufferTemp;
+	m_outputBufferRead = outputBufferTemp;
```

Comment:
You can use [`std::swap`](https://en.cppreference.com/w/cpp/algorithm/swap) instead.